Create MachineLearningTechnices.py

Alex-F-S opened the pull request:
Add imports for sklearn, numpy and pandas, as well as two functions to return a sklearn linear model, one with np input and the other with pd input

Alex-F-S committed:
Create MachineLearningTechnices.py

Add imports for sklearn, numpy and pandas, as well as two functions to return a sklearn linear model, one with np input and the other with pd input

Alex-F-S committed:
Update BaseLib.py

Añadir determinante, adjunta y inversa de una matriz nativa

Alex-F-S committed:
Update BaseLib.py

reb0rt081 commented:
Fes el codi i els comentaris en anglés i seguint el patró: MatrixDeterminant, MatrixTranspose, etc.

reb0rt081 commented:
Canviar nom de MachineLearningTechnices.py to MachineLearningTechniques.py

reb0rt081 commented:
Escriure un unit test per cada nou mètode (veure TODOs)

Commit:
Fixing some comments

Alex-F-S committed:
Update MachineLearningTechniques.py

Simplified model return function and added a function to return slope and intercept from linear model

Alex-F-S committed:
Update BaseLib.py

Updated namings for functions

Alex-F-S committed:
Update PythonTests.cs

Matrix unit test

Commit:
Refactoring code

reb0rt081 commented:
Hola Alex, el test del MatrixDeterminant no funciona per a matrius de rang 3 o superior. Revisa el codi!

Commit:
Fixing method

Commit:
Unit test for matrix inverse

Alex-F-S committed:
Update BaseLib.py

Changed determinant to use LU algorithm

Alex-F-S committed:
Update MachineLearningTechniques.py

Added k-means model and parameters functions

Alex-F-S committed:
Update MachineLearningTechniques.py

Added logistic regression model return and parameters return

Alex-F-S committed:
Update MachineLearningTechniques.py

Added SVC Model, Logistic Model, KMeans model, KNN model and evaluation function for any type of model

Alex-F-S committed:
Update PythonTests.cs

Added unit test for LinearModelParams from MachineLearningTechniques.py

Commit:
Setting  option to script

Commit:
Merge latest to Alex-Dev

Commit:
Removing unnecessary test

Alex-F-S committed:
Update python-build-test.yml

Update lib imports

Alex-F-S committed:
Update test_BaseLib.py

Added tests for DotProd, MatrixSum, MatrixMult and Determinant

Alex-F-S committed:
Update python-build-test.yml

Corrected scikit-learn import

Alex-F-S committed:
Update test_BaseLib.py

Re-test

Alex-F-S committed:
Update test_BaseLib.py

Added test for transpose and inverse

Alex-F-S committed:
Update test_BaseLib.py

Fixed error in assert

Alex-F-S committed:
Create test_MachineLearningTechniques.py

Added test file for MachineLearningTechniques with tests for all models

Alex-F-S committed:
Delete src/Python/test_MachineLearningTechniques.py

Deleted and moved to scripts foler

Alex-F-S committed:
Create test_MachineLearningTechniques.py

Added unit test for models in MachineLearningTechniques

Alex-F-S committed:
Update test_MachineLearningTechniques.py

Fixed X array dimensions

Alex-F-S committed:
Update test_MachineLearningTechniques.py

Revised asserts

Alex-F-S committed:
Update test_MachineLearningTechniques.py

Rev 2 for linear model and KNN

Alex-F-S committed:
Update test_MachineLearningTechniques.py

Changed to np array in linear model due to shape attribute missing

Alex-F-S committed:
Update test_MachineLearningTechniques.py

Added missing np import

Alex-F-S committed:
Update test_MachineLearningTechniques.py

Fixed error with numpy conversion

Alex-F-S committed:
Update test_MachineLearningTechniques.py

Fixed linear model intercept access

Alex-F-S committed:
Update MachineLearningTechniques.py

Added Ridge Model

Alex-F-S committed:
Update MachineLearningTechniques.py

Fixed comment

Alex-F-S committed:
Update test_MachineLearningTechniques.py

Fixed test for kmeans

Alex-F-S committed:
Update test_MachineLearningTechniques.py

Fixed typo with inertia

Alex-F-S committed:
Update MachineLearningTechniques.py

Fixed flattening on kmeansparams return of centroids

Alex-F-S committed:
Update test_MachineLearningTechniques.py

Serms that the kmeansparams function returns the centroids in reverse order

Alex-F-S committed:
Update test_MachineLearningTechniques.py

Fixed the classification test in KMeans Params

Alex-F-S committed:
Update test_MachineLearningTechniques.py

It seems sklearn doesnt mind the order of the centroids, so they are now ordered in ascending order and classifications cannot be tested, but under revision of the test outputs, they are correct

Alex-F-S committed:
Update MachineLearningTechniques.py

Update test
